### PR TITLE
Add alpine.3.16

### PR DESCRIPTION
**Pull request by raphael-proust**
Alpine 3.16 has been released a few months ago. This PR adds support for it.

### CODE REVIEW

File: src-opam/distro.ml [hunk 1]
```diff
@@ -542,6 +545,7 @@ let distro_status (d : t) : status =
         `Deprecated
     | `Alpine `V3_14 -> `Active `Tier2
     | `Alpine `V3_15 -> `Active `Tier1
+    | `Alpine `V3_16 -> `Active `Tier1
```
Comment (raphael-proust) [inline src-opam/distro.ml:548]: Note sure about this. I read the documentation on tiers but I still don't know if that's the correct tier to set for the alpine versions.

Comment (kit-ty-kate) [reply src-opam/distro.ml:548]: 3.15 should probably be downgraded to Tier2. Otherwise this would create all the variant images for both 3.15 and 3.16

**Review by avsm — APPROVED** [review]
lgtm once the comment is addressed. Let's get the CI running 3.16 as soon as possible as its been out for 4 months already.

On a broader subject, it might be worth pushing `alpine-edge` images as well (without putting them in CI, but building ocaml on it) so that we can catch bleeding edge issues in Alpine. It's become a pretty important distro for OCaml users these days, and we do already test debian-unstable in a similar fashion.

**Comment by tmcgilchrist** [comment]
Antonin’s pr adding it to the base image builder
https://github.com/ocurrent/docker-base-images/pull/190

**Pull request closed without merging**